src/generated/xproto.rs

```diff
@@ -6511,251 +7991,269 @@ pub struct CreateWindowAux {
+            colormap.serialize_into(bytes);
+        }
+        if let Some(cursor) = self.cursor {
+            bytes.reserve(4);
+            cursor.serialize_into(bytes);
+        }
     }
-    fn value_mask(&self) -> u32 {
-        let mut mask = 0;
+}
+impl CreateWindowAux {
+    #[allow(dead_code)]
+    fn switch_expr(&self) -> u32 {
+        let mut expr_value: u32 = 0;
         if self.background_pixmap.is_some() {
-            mask |= u32::from(CW::BackPixmap);
+            expr_value |= u32::from(Cw::BackPixmap);
         }
         if self.background_pixel.is_some() {
-            mask |= u32::from(CW::BackPixel);
+            expr_value |= u32::from(Cw::BackPixel);
         }
         if self.border_pixmap.is_some() {
-            mask |= u32::from(CW::BorderPixmap);
+            expr_value |= u32::from(Cw::BorderPixmap);
         }
         if self.border_pixel.is_some() {
-            mask |= u32::from(CW::BorderPixel);
+            expr_value |= u32::from(Cw::BorderPixel);
         }
         if self.bit_gravity.is_some() {
-            mask |= u32::from(CW::BitGravity);
+            expr_value |= u32::from(Cw::BitGravity);
         }
         if self.win_gravity.is_some() {
-            mask |= u32::from(CW::WinGravity);
+            expr_value |= u32::from(Cw::WinGravity);
         }
         if self.backing_store.is_some() {
-            mask |= u32::from(CW::BackingStore);
+            expr_value |= u32::from(Cw::BackingStore);
         }
         if self.backing_planes.is_some() {
-            mask |= u32::from(CW::BackingPlanes);
+            expr_value |= u32::from(Cw::BackingPlanes);
         }
         if self.backing_pixel.is_some() {
-            mask |= u32::from(CW::BackingPixel);
+            expr_value |= u32::from(Cw::BackingPixel);
         }
         if self.override_redirect.is_some() {
-            mask |= u32::from(CW::OverrideRedirect);
+            expr_value |= u32::from(Cw::OverrideRedirect);
         }
         if self.save_under.is_some() {
-            mask |= u32::from(CW::SaveUnder);
+            expr_value |= u32::from(Cw::SaveUnder);
         }
         if self.event_mask.is_some() {
-            mask |= u32::from(CW::EventMask);
+            expr_value |= u32::from(Cw::EventMask);
         }
         if self.do_not_propogate_mask.is_some() {
-            mask |= u32::from(CW::DontPropagate);
+            expr_value |= u32::from(Cw::DontPropagate);
         }
         if self.colormap.is_some() {
-            mask |= u32::from(CW::Colormap);
+            expr_value |= u32::from(Cw::Colormap);
         }
         if self.cursor.is_some() {
-            mask |= u32::from(CW::Cursor);
+            expr_value |= u32::from(Cw::Cursor);
         }
-        mask
+        expr_value
+    }
+}
+impl CreateWindowAux {
+    /// Create a new instance with all fields unset / not present.
+    pub fn new() -> Self {
+        Default::default()
     }
-    /// Set the background_pixmap field of this structure.
```

Comment:
I will not claim that this doc comment is great, but still... could you add it back?

Reply:
Done